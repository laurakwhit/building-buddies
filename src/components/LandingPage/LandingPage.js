import React, { Component } from 'react';
import PropTypes from 'prop-types';

import './LandingPage.scss';

import SignUpModal from '../SignUpModal/SignUpModal';

class LandingPage extends Component {
  state = {
    modalOpen: false
  };

  handleClick = () => {
    this.setState({ modalOpen: true });
  };

  handleModalClose = () => {
    this.setState({ modalOpen: false });
  };

  render() {
    const { modalOpen } = this.state;
    const { buildings, interests, userSignUp } = this.props;

    return (
      <div className="landing-page">
        {modalOpen && (
          <SignUpModal
            handleModalClose={this.handleModalClose}
            buildings={buildings}
            interests={interests}
            userSignUp={userSignUp}
          />
        )}
        <h1>Hello!</h1>
        <button onClick={this.handleClick}>Sign Up</button>
      </div>
    );
  }
}

LandingPage.propTypes = {
  setUser: PropTypes.func,
  buildings: PropTypes.array
};

export default LandingPage;
