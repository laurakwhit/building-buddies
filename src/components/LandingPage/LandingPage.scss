.landing-page {
  height: 100vh;
  position: relative;
}

h1 {
  margin-top: 0;
}
