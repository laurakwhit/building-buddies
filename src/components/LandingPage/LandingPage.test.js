import React from 'react';
import { shallow } from 'enzyme';

import LandingPage from './LandingPage';

describe('LandingPage', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<LandingPage />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
