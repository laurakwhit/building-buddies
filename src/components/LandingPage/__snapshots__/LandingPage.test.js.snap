// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LandingPage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        false,
        <h1>
          Hello!
        </h1>,
        <button
          onClick={[Function]}
        >
          Sign Up
        </button>,
      ],
      "className": "landing-page",
    },
    "ref": null,
    "rendered": Array [
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Hello!",
        },
        "ref": null,
        "rendered": "Hello!",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Sign Up",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Sign Up",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          false,
          <h1>
            Hello!
          </h1>,
          <button
            onClick={[Function]}
          >
            Sign Up
          </button>,
        ],
        "className": "landing-page",
      },
      "ref": null,
      "rendered": Array [
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Hello!",
          },
          "ref": null,
          "rendered": "Hello!",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sign Up",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Sign Up",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
